src/compiler/checker.ts

```diff
@@ -11246,7 +11246,20 @@ module ts {
 
         function checkGrammarFunctionLikeDeclaration(node: FunctionLikeDeclaration): boolean {
             // Prevent cascading error by short-circuit
-            return checkGrammarModifiers(node) || checkGrammarTypeParameterList(node, node.typeParameters) || checkGrammarParameterList(node.parameters);
+            return checkGrammarModifiers(node) || checkGrammarTypeParameterList(node, node.typeParameters) || checkGrammarParameterList(node.parameters) || checkGrammarArrowFunction(node);
+        }
+
+        function checkGrammarArrowFunction(node: FunctionLikeDeclaration): boolean {
+            if (node.kind === SyntaxKind.ArrowFunction) {
+                var arrowFunction = <ArrowFunction>node;
+                var sourceFile = getSourceFileOfNode(node);
+                var equalsGreaterThanLine = getLineAndCharacterOfPosition(sourceFile, getTokenPosOfNode(arrowFunction.equalsGreaterThanToken, sourceFile)).line;
+                var parametersLine = getLineAndCharacterOfPosition(sourceFile, arrowFunction.parameters.end).line;

```

Comment:
Swap `parametersLine` and `equalsGreaterThanLine` parameters; also use the `end`s of `arrowFunction.equalsGreaterThanToken,` since the full start and end are on the same line.